### ISSUE TITLE
Segmentation fault with specific `@media` and `@at-root` combinations

### ISSUE THREAD

**Issue by anlutro** [sass/libsass#859]
I'm using sassc (compiled from master today) and ran across a segmentation fault with a specific combination of `@media`, selectors and `@at-root`.

This is the minimal code I've been able to reproduce the segmentation fault in.

```sass
@media screen
	.two
		@at-root .one
			background: blue
			.three
				color: red
```

```
$ sassc ./test.sass 
Segmentation fault
```

Removing any bit of this example makes it compile just fine - it's something about the combination of rules (`background: blue`) and selectors (`.three`) within the `@at-root` declaration that's messing up.

**Comment by xzyfer** [comment]
I've confirmed this is an issue on the latest master

```scss
@media screen {
  .two {
    @at-root .one {
      background: blue;
      .three {
        color: red;
      }
    }
  }
}
```

**Comment by HugoGiraudel** [comment]
@xzyfer Does this mean I missed a case with my `@at-root` test suite?

**Comment by xzyfer** [comment]
It would appear so.

**Comment by xzyfer** [comment]
I've got a fix https://github.com/sass/libsass/pull/861.

I don't have time to create the spec now, but when I do this is ready to go.

**Comment by xzyfer** [comment]
Spec added https://github.com/sass/sass-spec/pull/248

**Comment by xzyfer** [comment]
This is fixed and will be 3.2.

**Comment by anlutro** [comment]
Awesome. I love using sassc, keep up the great work :)
